### PR TITLE
Gemini 1.5 multimodal snippets

### PR DESCRIPTION
**Pull request by glaforge**
Fixes b/330496709, b/330500930, b/330497454, b/330500054, adding multimodal snippets using Gemini 1.5

### CODE REVIEW

File: vertexai/snippets/src/main/java/vertexai/gemini/MultimodalAllInput.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-google-cloud-project-id";
+    String location = "us-central1";
+    String modelName = "gemini-1.5-pro-preview-0409";
+
+    multimodalAllInput(projectId, location, modelName);
+  }
+
+  // Analyzes the given audio input.
```
Comment (Sita04) [inline vertexai/snippets/src/main/java/vertexai/gemini/MultimodalAllInput.java:40]: Should this description be updated?

File: vertexai/snippets/src/main/java/vertexai/gemini/MultimodalVideoAudioInput.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package vertexai.gemini;
+
+// [START generativeaionvertexai_gemini_video_with_audio]
```
Comment (Sita04) [inline vertexai/snippets/src/main/java/vertexai/gemini/MultimodalVideoAudioInput.java:19]: Kindly update this region tag, as its also used in `MultimodalAllInput`

File: vertexai/snippets/src/main/java/vertexai/gemini/MultimodalVideoAudioInput.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-google-cloud-project-id";
+    String location = "us-central1";
+    String modelName = "gemini-1.5-pro-preview-0409";
+
+    videoAudioInput(projectId, location, modelName);
+  }
+
+  // Analyzes the given audio input.
```
Comment (Sita04) [inline vertexai/snippets/src/main/java/vertexai/gemini/MultimodalVideoAudioInput.java:40]: Comment should be updated to reflect the method.

File: vertexai/snippets/src/test/java/vertexai/gemini/SnippetsIT.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2023 Google LLC
+ * Copyright 2024 Google LLC
```
Comment (Sita04) [inline vertexai/snippets/src/test/java/vertexai/gemini/SnippetsIT.java:2]: The copyright year refers to when the file was initially created. Hence, please revert the year.

**Review by Sita04 — CHANGES_REQUESTED** [review]
Thanks for the snippets!

File: vertexai/snippets/src/main/java/vertexai/gemini/MultimodalAllInput.java (hunk 1 above)
Comment (glaforge) [reply vertexai/snippets/src/main/java/vertexai/gemini/MultimodalAllInput.java:40]: I'll put `A request containing a text prompt, a video, and a picture.`

File: vertexai/snippets/src/test/java/vertexai/gemini/SnippetsIT.java (hunk 4 above)
Comment (glaforge) [reply vertexai/snippets/src/test/java/vertexai/gemini/SnippetsIT.java:2]: Good to know, I'll revert.

File: vertexai/snippets/src/main/java/vertexai/gemini/MultimodalVideoAudioInput.java (hunk 2 above)
Comment (glaforge) [reply vertexai/snippets/src/main/java/vertexai/gemini/MultimodalVideoAudioInput.java:19]: Well spotted. Copy and paste error. It's `generativeaionvertexai_gemini_all_modalities` I should have used in `MultimodalAllInput`.

File: vertexai/snippets/src/main/java/vertexai/gemini/MultimodalVideoAudioInput.java (hunk 3 above)
Comment (glaforge) [reply vertexai/snippets/src/main/java/vertexai/gemini/MultimodalVideoAudioInput.java:40]: I'll put `Analyzes the given video input, including its audio track.` which is more explicit.

**Review by gericdong — APPROVED** [review]

**Review by Sita04 — APPROVED** [review]

**Comment by Sita04** [comment]
@glaforge Requesting to wait until this PR has been merged:https://github.com/GoogleCloudPlatform/java-docs-samples/pull/9303

**Pull request merged**